services/backend/src/core/profile/util/updateProfile.js

```diff
@@ -148,14 +148,93 @@ async function updateProfile(request, userId, language) {
 
   // Deletes every experiences and educations if experiences or educations is defined since
   // there's no way to uniquely identify them solely from the data
+  let deleteAll = [];
+
   if (experiences) {
-    await prisma.experience.deleteMany({ where: { userId } });
+    const attachmentLinkId = flatten(
+      await Promise.all(
+        experiences.map((ed) =>
+          prisma.attachmentLink.findMany({
+            where: { experienceId: ed.id },
+            select: { id: true },
+          })
+        )
+      )
+    );
```

Comment:
I'm curious, what is the use of `flatten`?

Reply:
The structure of this method was kind of confusing. So the return of this method was a nested array of depth 1 which made it hard to check for the length.